src/edu/stanford/nlp/sempre/Formula.java

```diff
@@ -24,6 +24,10 @@
   public abstract LispTree toLispTree();
 
   // Recursively perform some operation on each formula.
+  // Apply to formulas.  If |func| returns false, then recurse on children.
+  public abstract void recurse(Function<Formula, Boolean> func);
```

Comment:
Unlike map, this function does not construct a new object, which should make it a little more efficient. I have been using it in my code.